src/test/fuzz/uint256.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2020-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <uint256.h>
+#include <util/transaction_identifier.h>
+
+FUZZ_TARGET(uint256_fromhex)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    const auto input{fuzzed_data_provider.ConsumeRandomLengthString()};
+    const auto result{uint256::FromHex(input)};
```

Comment:
Strikes me as being "too far off the reservation" with the `input`s. Won't we be hitting the first condition in `FromHex` on `str.size()` being different from `::size() * 2` in >99% of cases? Maybe something closer to the below would spend fuzz-cycles a bit more wisely? Also added a loop to use more of the fuzz data. Then again I'm new to fuzzing.
```diff
diff --git a/src/test/fuzz/uint256.cpp b/src/test/fuzz/uint256.cpp
index f5c8b2e146..38bbc5e108 100644
--- a/src/test/fuzz/uint256.cpp
+++ b/src/test/fuzz/uint256.cpp
@@ -9,15 +9,42 @@
 
 FUZZ_TARGET(uint256_fromhex)
 {
+    enum Mode {
+        RandomCharsRandomLength,
+        RandomCharsCorrectLength,
+        ValidCharsRandomLength,
+
+        kMaxValue = ValidCharsRandomLength
+    };
+
     FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
-    const auto input{fuzzed_data_provider.ConsumeRandomLengthString()};
-    const auto result{uint256::FromHex(input)};
-    if (result) {
-        // we only expect a result for a 64 character hex-string
-        assert(input.length() == 64);
+    while (fuzzed_data_provider.remaining_bytes() > 0) {
+        std::string input;
+        static_assert(64 == uint256::size() * 2, "2 hex chars per byte");
+        switch (fuzzed_data_provider.ConsumeEnum<Mode>()) {
+        case RandomCharsRandomLength:
+            input = fuzzed_data_provider.ConsumeRandomLengthString();
+            break;
+        case RandomCharsCorrectLength:
+            input = fuzzed_data_provider.ConsumeBytesAsString(64);
+            break;
+        case ValidCharsRandomLength:
+            constexpr char valid[] = "0123456789abcdefABCDEF";
+            input.resize(64, '0');
+            for (char& c : input) {
+                c = valid[fuzzed_data_provider.ConsumeIntegralInRange<int>(0, sizeof(valid)-1)];
+            }
+            break;
+        }
+
+        const auto result{uint256::FromHex(input)};
+        if (result) {
+            // we only expect a result for a 64 character hex-string
+            assert(input.length() == 64);
 
-        // check we can also construct {T,Wt}xids from a valid 64 character hex-string input
-        assert(Txid::FromHex(input));
-        assert(Wtxid::FromHex(input));
+            // check we can also construct {T,Wt}xids from a valid 64 character hex-string input
+            assert(Txid::FromHex(input));
+            assert(Wtxid::FromHex(input));
+        }
     }
-}
\ No newline at end of file
+}
```

Reply:
I'm lacking hands-on experience with fuzzing to comment on worthwhile optimizations, so I'm going to go with consensus here. @maflcko's reasoning looks sounds to me, so I think keeping it as is (besides moving to to `fuzz/hex.cpp` as discussed in another comment) is the sensible approach, but thank you for sharing the `Enum` approach, I hadn't seen that yet and it looks like a useful pattern to know.